File: Sorting/merge_two_sorted_array.java
package Sorting;
import java.util.Scanner;

public class merge_two_sorted_array {
    public static void main (String[] args){
        Scanner scn = new Scanner(System.in);
        int n1 = scn.nextInt();
        int [] arr1 = new int[n1];
        for (int i=0; i<n1; i++){
            arr1[i] = scn.nextInt();
        }
        int n2 = scn.nextInt();
        int [] arr2 = new int[n2];
        for(int i =0; i<n2; i++){
            arr2[i] = scn.nextInt();
        }
        scn.close();
        merge_array(arr1, arr2);
    } 

    public static void merge_array(int [] arr1, int [] arr2){
        int i =0;
        int j = 0;
        int k = 0;
        int [] res = new int [arr1.length + arr2.length];
        while (i < arr1.length && j < arr2.length){
            if (arr1[i] < arr2[j]){
                res[k] = arr1[i];
                i++;
                k++; 
            }
            else{
                res[k] = arr2[j];
                j++;
                k++;
            }
        }
        if(i == arr1.length){
            while(j != arr2.length){
                res[k] = arr2[j];
                j++;
                k++;
            }
        }
        if(j == arr2.length){
            while(i!=arr1.length){
                res[k] = arr1[i];
                i++;
                k++;
            } 
        }

        System.out.print("Merged Two Sorted Array : ");
        for (int p = 0 ; p <res.length; p++){
            System.out.print(res[p] + "  ");
        }
    }  
}

File: Sorting/merge_two_sorted_array.py
def mergearray(arr1, arr2):
    i = 0
    j = 0
    k = 0
    res = [len(arr1) + len(arr2)]
    while(i < len(arr1) and j<len(arr2)):
        print(i, j, k, res)
        if arr1[i]<arr2[j]:
            print("In If")
            res[k] = arr1[i]
            i+=1
            k+=1
        else:
            print("In Else")
            res[k] = arr2[j]
            j+=1
            k+=1
    
    if (i != len(arr1)):
        while(i!= len(arr1)):
            res[k] = arr1[i]
            i+=1
            k+=1
        
    if (j != len(arr2)):
        while( j!= len(arr2)):
            res[k] = arr2[j]
            j+=1
            k+=1
    
    return res

    
n1 = int(input())
arr1 = [None]*n1
for i in range(n1):
    arr1[i] = int(input())

n2 = int(input())
arr2 = [None]*n2
for i in range(n2):
    arr2[i] = int(input())

print(mergearray(arr1, arr2))